metrics/web/lib/common/presentation/graphs/circle_percentage.dart

```diff
@@ -1,53 +1,65 @@
 
 /// The widget that represents the metric percent in a circular graph.
 ///
 /// If no constraints are given by the parent widget, it will be as big as possible.
 /// Otherwise, it will match its parent's size.
 class CirclePercentage extends StatefulWidget {
+  /// The percent value of the metric.
   final double value;
+
+  /// The padding of the [value] text inside the circle graph.
   final EdgeInsets padding;
+
+  /// The wight of the graph's stroke.
   final double strokeWidth;
+
+  /// The color of the part of the graph that represents the value.
   final Color valueColor;
+
+  /// The color of the graph's circle itself.
   final Color strokeColor;
+
+  /// The color used to fill the graph.
   final Color backgroundColor;
+
+  /// The [TextStyle] of the percent text.
   final TextStyle valueStyle;
+
+  /// The width of the value (filled) stroke.
   final double valueStrokeWidth;
 
-  /// Creates the circle graph.
+  /// The [Widget] displayed instead of [value] if the [value] is `null`.
+  final Widget placeholder;
+
+  /// Creates the circle percentage graph.
   ///
-  /// The [value] must be `null` or from 0.0 (inclusive) to 1.0 (inclusive).
+  /// The [value] must be either `null` or from 0.0 (inclusive) to 1.0 (inclusive).
+  /// The [placeholder] must not be `null`. The [placeholder] is displayed,
+  /// if the [value] is `null`.
   ///
-  /// [value] is the percent value of the metric.
-  /// If the [value] is null, the [DashboardStrings.noDataPlaceholder] is displayed.
-  /// [padding] is the padding of the [value] text inside the circle graph.
-  /// [strokeWidth] is the wight of the graph's stroke. Defaults to 2.0.
-  /// [valueColor] is the color of the part of the graph that represents the value.
-  /// If nothing is passed, the [MetricWidgetThemeData.primaryColor] is used.
-  /// [strokeColor] is the color of the graph's circle itself.
-  /// If nothing is passed, the [MetricWidgetThemeData.accentColor] is used.
-  /// [backgroundColor] is the color to fill the graph.
-  /// If nothing is passed, the [MetricWidgetThemeData.backgroundColor] is used.
-  /// [valueStyle] is the [TextStyle] of the percent text.
-  /// [valueStrokeWidth] is the width of the value (filled) stroke. Defaults to 5.0.
+  /// The [strokeWidth] default value is 2.0.
+  /// The [valueColor] default value is [Colors.blue].
+  /// The [strokeColor] default value is [Colors.grey].
+  /// The [valueStrokeWidth] default value is 5.0.
   const CirclePercentage({
     Key key,
     @required this.value,
+    @required this.placeholder,
     this.valueStyle,
     this.strokeWidth = 2.0,
     this.valueStrokeWidth = 5.0,
     this.padding = EdgeInsets.zero,
-    this.valueColor,
-    this.strokeColor,
+    Color valueColor,
+    Color strokeColor,
     this.backgroundColor,
-  })  : assert(value == null || (value >= 0 && value <= 1)),
+  })  : assert(placeholder != null),
+        assert(value == null || (value >= 0 && value <= 1)),
```

Comment:
Should we be more fail-proof here? As we use ViewModels now they don't guarantee that the value is within those bounds - so what if this widget can accept any double - but if it's not within bounds - the value is pushed to be within bounds, like -1 -> 0 and values more than 1 are treated as 1? Should we discuss it a bit more?